src/auxiliary/auxiliary.jl

```diff
@@ -225,6 +225,20 @@ macro threaded(expr)
                 end
             end
         end
+    elseif _PREFERENCE_THREADING === :dynamic
+        quote
+            let
+                if $Threads.nthreads() == 1
+                    $(expr)
+                else
+                    $Threads.@threads :dynamic $(expr)
+                end
+            end
+        end
```

Comment:
This will introduce bugs for code like

https://github.com/trixi-framework/Trixi.jl/blob/f35d749f57c2d933941a39834f5833bfbdadb36a/src/solvers/dgsem_tree/dg_2d.jl#L945-L950

won't it? @vchuravy

Reply:
So I think we will need to remove the "dynamic" option for now from this PR.